data-access-layer/eventmesh/LockManager.js

```
@@ -167,28 +172,45 @@ class LockManager {
+  unlock(resourceId, lockId, maxRetryCount, retryDelay) {
     assert.ok(resourceId, `Parameter 'resourceId' is required to release lock`);
-    maxRetryCount = maxRetryCount || CONST.ETCD.MAX_RETRY_UNLOCK;
+    // TODO-PR: making lockId not mendatory as currently we don't have deployment resources
+    // hence from lastOperation call we can't pass lockId for unlock call
+    // assert.ok(lockId, `Parameter 'lockId' is required to release lock`);
+    maxRetryCount = maxRetryCount || CONST.APISERVER.MAX_RETRY_UNLOCK;
     retryDelay = retryDelay || CONST.APISERVER.RETRY_DELAY;
     logger.debug(`Attempting to unlock resource ${resourceId}`);
     return utils.retry(tries => {
       logger.info(`+-> Attempt ${tries + 1} to unlock resource ${resourceId}`);
-      return eventmesh.apiServerClient.deleteResource({
-          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.LOCK,
-          resourceType: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT_LOCKS,
-          resourceId: resourceId
-        })
+      const opts = {
+        resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.LOCK,
+        resourceType: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT_LOCKS,
+        resourceId: resourceId,
+        status: {
+          state: CONST.APISERVER.RESOURCE_STATE.UNLOCKED
+        }
+      };
+      if (lockId) {
+        opts.metadata = {
+          resourceVersion: lockId
+        };
+      }
+      return eventmesh.apiServerClient.updateResource(opts)
         .tap(() => logger.debug(`Successfully unlocked resource ${resourceId} `))
         .catch(err => {
+          if (err instanceof Conflict) {
+            logger.info(`Lock on resource ${resourceId} has been updated by some other operation because it expired, no need to unlock now`, err);
+            return;
+          }
           if (err instanceof NotFound) {
-            logger.debug(`Successfully Unlocked resource ${resourceId} `);
+            logger.debug(`Lock on resource ${resourceId} not found`, err);
```

When would this happen in the new flow? This error should never occur right as lock will always be present? So better to throw this error in this scenario?